api/storage/client_test.go

```diff
@@ -145,3 +145,54 @@ func (s *storageMockSuite) TestCreatePool(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(called, jc.IsTrue)
 }
+
+func (s *storageMockSuite) TestListVolumes(c *gc.C) {
+	var called bool
+	machines := []string{"one", "two"}
+	apiCaller := basetesting.APICallerFunc(
+		func(objType string,
+			version int,
+			id, request string,
+			a, result interface{},
+		) error {
+			called = true
+			c.Check(objType, gc.Equals, "Storage")
+			c.Check(id, gc.Equals, "")
+			c.Check(request, gc.Equals, "ListVolumes")
+
+			args, ok := a.(params.StorageVolumeFilter)
+			c.Assert(ok, jc.IsTrue)
+			c.Assert(args.Machines, gc.HasLen, 2)
+
+			if results, k := result.(*params.StorageVolumesResult); k {
+				disks := make([]params.StorageDisk, len(args.Machines))
+				for i, m := range args.Machines {
+					attachment := params.StorageAttachment{
+						Tag:         fmt.Sprintf("tag%v", i),
```

Comment:
I'm not sure volume attachments will have tags. They're uniquely identified by (storage-ID, machine-ID). Can you please leave this out?